lib/internal/modules/esm/worker.js

```diff
@@ -0,0 +1,25 @@
+const { workerData } = require('internal/worker_threads');
+const { ESMLoader } = require('internal/modules/esm/loader');
+
+const { comsChannel } = workerData;
+const lock = new Int32Array(comsChannel, 0, 4); // Required by Atomics
+const requestResponseData = new Uint8Array(comsChannel, 4, 2044); // for TextEncoder/Decoder
+
+const publicESMLoader = new ESMLoader();
+
+Atomics.store(lock, 0, 0); // Send 'ready' signal to main
+Atomics.notify(lock, 0); // Notify main of signal
+
+while (true) { // event loop
+  Atomics.wait(lock, 0, 0); // This pauses the while loop
+  // Worker is now active and main is sleeping
+  const { data, type } = JSON.parse(
+    (new TextDecoder().decode(requestResponseData))
+      .replaceAll('\x00', '') // strip empty space (not a great solution)
```

Comment:
What’s wrong with this solution, and what would be a better one?

Reply:
The empty space should just be ignored. Guy suggested using another array to achieve it, but I haven't looked into that (was gonna do in polish—this gets the job done).

Reply:
Can you explain the problem a bit more? Why is there unwanted empty space in the first place, and what problem does it cause?

Reply:
Because a SharedArrayBuffer has a fixed size (so it's pre-allocated to be quite large). When you put a pebble into a bucket, there's a lot of empty space. That's what's happening here.